src/eva/vision/data/datasets/segmentation/consep.py

```diff
@@ -0,0 +1,167 @@
+"""PANDA dataset class."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, List, Literal, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from torchvision import tv_tensors
+from torchvision.transforms.v2 import functional
+from typing_extensions import override
+
+from eva.vision.data.datasets import _validators, wsi
+from eva.vision.data.datasets.segmentation import base
+from eva.vision.data.wsi.patching import samplers
+from eva.vision.utils import io
+
+
+class CoNSeP(wsi.MultiWsiDataset, base.ImageSegmentation):
+    """Dataset class for CoNSeP semantic segmentation task.
+
+    We combine classes 3 (healthy epithelial) & 4 (dysplastic/malignant epithelial)
+    into the epithelial class and 5 (fibroblast), 6 (muscle) & 7 (endothelial) into
+    the spindle-shaped class.
+    """
+
+    _expected_dataset_lengths: Dict[str | None, int] = {
+        "train": 27,
+        "val": 14,
+        None: 41,
+    }
+
+    def __init__(
+        self,
+        root: str,
+        sampler: samplers.Sampler,
+        split: Literal["train", "val"] | None = None,
+        width: int = 224,
+        height: int = 224,
+        target_mpp: float = 0.25,
+        transforms: Callable | None = None,
+        seed: int = 42,
```

Comment:
I don't think we need a seed here, since we fix the seed at the start of the experiments - if it is for unit tests, there are ways to fix it there

Reply:
removed